bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/converter/type/DateTimeTypeConverter.java

```diff
@@ -0,0 +1,119 @@
+
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.util.Locale;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.siemenshvac.internal.converter.ConverterException;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.types.Type;
+
+import com.google.gson.JsonElement;
+
+/**
+ * Converts between a SiemensHvac datapoint value and an openHAB DecimalType.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+public class DateTimeTypeConverter extends AbstractTypeConverter {
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    public DateTimeTypeConverter(final TimeZoneProvider timeZoneProvider) {
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    protected boolean toBindingValidation(Type type) {
+        return true;
+    }
+
+    @Override
+    protected @Nullable Object toBinding(Type type, ChannelType tp) throws ConverterException {
+        Object valUpdate = null;
+
+        if (type instanceof DateTimeType dateTime) {
+            valUpdate = dateTime.toString();
+        }
+
+        return valUpdate;
+    }
+
+    @Override
+    protected boolean fromBindingValidation(JsonElement value, String type) {
+        return true;
+    }
+
+    @Override
+    protected DateTimeType fromBinding(JsonElement value, String type, ChannelType tp) throws ConverterException {
+        if ("----".equals(value.getAsString())) {
+            return new DateTimeType(ZonedDateTime.now(this.timeZoneProvider.getTimeZone()));
+        } else {
+            String[] formats = { "EEEE, d. LLLL yyyy HH:mm[:ss]", "d. LLLL yyyy HH:mm[:ss]", "d. LLLL yyyy",
+                    "d. LLLL" };
+
+            for (int i = 0; i < formats.length; i++) {
+                try {
+                    DateTimeFormatterBuilder formatterBuilder = new DateTimeFormatterBuilder().parseCaseInsensitive()
+                            .appendPattern(formats[i]).parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+                            .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+                            .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0);
+
+                    if (i == 3) {
+                        formatterBuilder = formatterBuilder.parseDefaulting(ChronoField.YEAR,
+                                ZonedDateTime.now(this.timeZoneProvider.getTimeZone()).getYear());
+                    }
+
+                    LocalDateTime parsedDate = LocalDateTime.parse(value.getAsString(),
+                            formatterBuilder.toFormatter(Locale.FRENCH));
```

Comment:
Is this product only for the French market, or why do you need hardcoded French Locale here? Can you configure the OZW webserver language - if yes, how will such a change affect this method?

Reply:
No it's a mistake, the product is international.

And you're right, changing the language on gateway side will influenced the result.
Mainly on formatter like this one "EEEE, d. LLLL yyyy HH:mm[:ss]" where the first part EEEE will be localized regarding the language setting of the gateway.

I take a look on how i can Handle this a better way

Reply:
I've modify the code of TypeConverter to add the Local information as a parameter of fromBinding function.
This information is take from the MetadataRegistry which already contains the current user used to connect the API.
I handle there the language retrivial, and locale initialization.

Current gateway is supporting a bunch of different language: English,Deutsch,Francais,Italiano,Nederlands,Polski,Cesky,Magyar,Espanol,Dansk,Svenska,Suomi,Portugues,Russkij,Turkce,Slovensky;

I will certainly need to do more test to verify that formatter are ok for all this language.

Reply:
Wow, that's a lot of languages to support. This reminds me of writing an integration for the old Miele@home Homebus API where essential information was provided as localized strings. It was a tremendous work to support that for the 4-5 languages supported by the gateway.

You might want to add to the README that depending on the configured language, there might be some issues which can then be reported (I suppose). You might want to prioritize to fully test and support at least a few of those languages besides French, for example English and German (considering the home of both Siemens and openHAB 😄).

Reply:
Add a paragraph inside the Readme.md about languages support.